src/components/analyzeModal/getFrameworksFromSaveFile.ts

```diff
@@ -14,12 +14,18 @@ export const getFrameworksFromSaveFile = ({
     if (weapon === undefined) {
       return;
     }
-    const frameworks = frameworkIDs
-      .map(
-        (frameworkID) =>
-          flipObject(FrameworkIDs[weapon])[frameworkID] as Framework | undefined
-      )
-      .filter((f) => f !== undefined) as Framework[];
+
+    const all_weapon_frameworks = FrameworkIDs[weapon];
+
+    const frameworks: Framework[] = [];
+    for (const frameworkID of frameworkIDs) {
+      for (const [framework, id] of Object.entries(all_weapon_frameworks)) {
+        if (id === frameworkID) {
+          frameworks.push(framework as Framework)
+        }
+      }
+    }
```

Comment:
You can probably optimize this a bit:

```suggestion
    for (const [framework, id] of Object.entries(all_weapon_frameworks)) {
      if (frameworkIDs.includes(id)) {
        frameworks.push(framework as Framework);
      }
    }
```

I think it's the same worst-case complexity but the `includes` should be more performant than the nested loop and will short-circuit